macros/misc/randomNamesPronouns.pl

```diff
@@ -0,0 +1,554 @@
+	'Martha'    => 'she',
+	'Matthew'   => 'he',
+	'Matty'     => 'they',
+	'Max'       => 'they',
+	'Maygen'    => 'she',
+	'Michael'   => 'he',
+	'Michele'   => 'she',
+	'Morah'     => 'she',
+	'Nathan'    => 'he',
+	'Neil'      => 'he',
+	'Nenia'     => 'she',
+	'Nicholas'  => 'he',
+	'Nina'      => 'she',
+	'Olivia'    => 'she',
+	'Page'      => 'she',
+	'Parnell'   => 'he',
+	'Penelope'  => 'she',
+	'Perlia'    => 'she',
+	'Peter'     => 'he',
+	'Phil'      => 'he',
+	'Priscilla' => 'she',
+	'Randi'     => 'he',
+	'Ravi'      => 'he',
+	'Ray'       => 'they',
+	'Rebecca'   => 'she',
+	'Renee'     => 'she',
+	'Rita'      => 'she',
+	'Ronda'     => 'she',
+	'Ross'      => 'he',
+	'Ryan'      => 'he',
+	'Samantha'  => 'she',
+	'Sarah'     => 'she',
+	'Scot'      => 'he',
+	'Sean'      => 'he',
+	'Sebastian' => 'he',
+	'Selena'    => 'she',
+	'Shane'     => 'he',
+	'Sharell'   => 'she',
+	'Sharnell'  => 'she',
+	'Sherial'   => 'she',
+	'Stephanie' => 'she',
+	'Stephen'   => 'he',
+	'Subin'     => 'she',
+	'Sydney'    => 'she',
+	'Tammy'     => 'she',
+	'Teresa'    => 'she',
+	'Thanh'     => 'he',
+	'Tien'      => 'he',
+	'Tiffany'   => 'she',
+	'Timothy'   => 'he',
+	'Tracey'    => 'she',
+	'Virginia'  => 'she',
+	'Wendy'     => 'she',
+	'Wenwu'     => 'he',
+	'Will'      => 'he'
+};
+
+@last_names = qw(
+	Adams   Allen     Alvarez    Anderson Bailey  Baker    Bennet    Brooks
+	Brown   Campbell  Carter     Castillo Chavez  Clark    Collins   Cook
+	Cooper  Cox       Cruz       Davis    Diaz    Edwards  Evans     Flores
+	Foster  Garcia    Gomez      Gonzales Gray    Green    Gutierrez Hall
+	Harris  Hernandez Hill       Howard   Hughes  Jackson  James     Jimenez
+	Johnson Jones     Kelly      Kim      King    Lee      Lewis     Long
+	Lopez   Martin    Martinez   Mendoza  Miller  Mitchell Moore     Morales
+	Morgan  Morris    Murphy     Myers    Nelson  Nguyen   Ortiz     Parker
+	Patel   Perez     Peterson   Phillips Pina    Price    Ramirez   Ramos
+	Reed    Reyes     Richardson Rivera   Roberts Robinson Rodriguez Rogers
+	Ross    Ruiz      Sanchez    Sanders  Scott   Smith    Stewart   Sullivan
+	Taylor  Thomas    Thompson   Torres   Turner  Walker   Ward      Watson
+	White   Williams  Wilson     Wood     Wright  Young
+);
+
+# need to have the keys sorted to have a consistent set of names for a given seed, so
+# we build an array of the first_names sorted.
+my @first_names;
+
+sub _randomNamesPronouns_init {
+	@first_names = lex_sort(keys(%$names_pronouns));
+}
+
+=head2 randomPerson
+
+Returns a person as a Person object from a list in the macro.
+
+=head3 Examples and options
+
+=over
+
+=item * No arguments returns a single random person
+
+Example
+
+  randomPerson()
+
+=item * C<n =E<gt> k> returns an array of k Person objects with unique names.
+
+Example
+
+  randomPerson(n=>5)
+
+returns an array of 5 Person objects with unique names.
+
+=item * C<names =E<gt> arrayref> returns an array of specified Person objects.
```

Comment:
Should this just say something about replacing the default list of names/pronouns with your own list? It does not seem like passing a `names` option should be described as returning an array, if n is 1.

Reply:
fixed.